jbmc/regression/jbmc/exception-cleanup/Test.java

```diff
@@ -0,0 +1,44 @@
+public class Test {
+
+  public static void main(int unknown) {
+
+    try {
+      mayThrow(unknown % 7 == 5);
+    }
+    catch(Exception e) {
+    }
+
+    // This test checks that symex can tell there is no exception in flight
+    // (i.e. @inflight_exception is null) at this point, requiring it to
+    // converge the `if @inflight_exception == null` test on mayThrow's return
+    // with the explicit `@inflight_exception = null;` assignment that concludes
+    // the catch block.
+
+    // If it knows that, it will also know the following catch block is
+    // unreachable; if not it will pursue both paths.
+    try {
+      mayThrow(false);
+    }
+    catch(Exception e) {
+      unreachable();
+    }
+
+    // Try it once more, to check we also know after an unreachable catch that
+    // there is still no inflight exception
```

Comment:
:bulb: A test with nested try-catch would be useful to prove that point.

Reply:
Such a case actually does not work yet, because symex can't tell that all exceptions were caught (i.e. the catch dispatch table is exhaustive). The reason is it isn't yet smart enough to notice that upon entering the catch block the incoming pointer cannot be null, and therefore e instanceof Exception certainly succeeds (null instanceof A returns false for any A).

Fixing that is my next task!

Reply:
You could add a KNOWNBUG test for it now and turn it on as part of your next task.